src/stancjs/stancjs.ml

```diff
@@ -22,29 +45,44 @@ let stan2cpp model_name model_string =
           "Semantic check failed but reported no errors. This should never \
            happen."
   in
-  Result.bind ast
-    ~f:
-      (Fn.compose semantic_err_to_string Semantic_check.semantic_check_program)
-  |> Result.map ~f:(fun typed_ast ->
-         let mir = Ast_to_Mir.trans_prog model_name typed_ast in
-         let tx_mir = Transform_Mir.trans_prog mir in
-         let cpp = Fmt.strf "%a" Stan_math_code_gen.pp_prog tx_mir in
-         cpp )
+  if is_flag_set "--version" then Result.Ok (Fmt.strf "%s" version, [])
+  else
+    Result.bind ast
+      ~f:
+        (Fn.compose semantic_err_to_string
+           Semantic_check.semantic_check_program)
+    |> Result.map ~f:(fun typed_ast ->
+           let mir = Ast_to_Mir.trans_prog model_name typed_ast in
+           let tx_mir = Transform_Mir.trans_prog mir in
+           let opt_mir =
+             if is_flag_set "--O" then Optimize.optimization_suite tx_mir
+             else tx_mir
+           in
+           let cpp = Fmt.strf "%a" Stan_math_code_gen.pp_prog opt_mir in
+           if is_flag_set "--warn-uninitialized" then
+             Pedantic_analysis.print_warn_uninitialized mir ;
+           if is_flag_set "--warn-pedantic" then
+             Pedantic_analysis.print_warn_pedantic mir ;
+           (cpp, []) )
 
 let wrap_result = function
-  | Result.Ok s ->
+  | Result.Ok (s, k) ->
       Js.Unsafe.obj
         [| ("result", Js.Unsafe.inject (Js.string s))
-         ; ("warnings", Js.Unsafe.inject Js.array_empty) |]
+         ; ( "warnings"
+           , Js.Unsafe.inject
+               (Js.array (List.to_array (List.map ~f:Js.string k))) ) |]
   | Error e ->
       Js.Unsafe.obj
         [| ("errors", Js.Unsafe.inject (Array.map ~f:Js.string [|e|]))
          ; ("warnings", Js.Unsafe.inject Js.array_empty) |]
 
-let map2 f (x, y) = (f x, f y)
-
-let wrap2 f s1 s2 =
-  let s1, s2 = map2 Js.to_string (s1, s2) in
-  f s1 s2 |> wrap_result
+let stan2cpp_wrapped name code (flags : Js.string_array Js.t Js.opt) =
+  stan2cpp (Js.to_string name) (Js.to_string code)
+    Js.(
+      Opt.map flags (fun a ->
+          str_array a |> Js.to_array |> Array.map ~f:to_string )
+      |> Opt.to_option)
+  |> wrap_result
 
-let () = Js.export "stanc" (wrap2 stan2cpp)
+let _ = Js.export "stanc" stan2cpp_wrapped
```

Comment:
Woops, I think this should stay `let () = ` no? The style guide doesn't like `_` unless necessary and then with type annotations (e.g. `let (_ : unit) =` (but `let () = ` is more concise).